django_usaid_clinicfinder/clinicfinder/tasks.py

```diff
@@ -0,0 +1,197 @@
+from __future__ import absolute_import
+
+from celery.task import Task
+from celery.utils.log import get_task_logger
+from celery.exceptions import SoftTimeLimitExceeded
+from suds.client import Client
+from django.contrib.gis.geos import Point
+from django.contrib.gis.measure import Distance
+from go_http.send import HttpApiSender
+
+from django.conf import settings
+
+from clinicfinder.models import (LBSRequest, LookupPointOfInterest,
+                                 LookupLocation, PointOfInterest,
+                                 Location)
+
+logger = get_task_logger(__name__)
+
+
+class LBS_Lookup(Task):
+
+    """
+    Task to request location from LBS API
+    """
+    name = "clinicfinder.tasks.lbs_lookup"
+
+    class FailedEventRequest(Exception):
+
+        """
+        The attempted task failed because of a non-200 HTTP return
+        code.
+        """
+
+    def lbs_api_client(self):
+        return Client(settings.LBS_API_WSDL)
+
+    def run(self, lbsrequest_id, **kwargs):
+        """
+        Gets user details, adds to whitelist and gets current location
+        """
+        l = self.get_logger(**kwargs)
+
+        l.info("Processing new LBS lookup")
+        response = "No response"
+        try:
+            lbsrequest = LBSRequest.objects.get(pk=lbsrequest_id)
+            client = self.lbs_api_client()
+            whitelist = client.service.AddAllowedMsisdn(
+                username=settings.LBS_API_USERNAME,
+                password=settings.LBS_API_PASSWORD,
+                msisdn=lbsrequest.search["msisdn"], permissionType=2)
+            if whitelist[0][0]["_code"] != "101":
+                response = whitelist[0][0]["_message"]
+                l.info("Failed to Add MSISDN to allowed list")
+                lbsrequest.response = {
+                    "whitelist_message": whitelist[0][0]["_message"]
```

Comment:
Should we add ``whitelist_code`` and maybed some sort of explicite ``"success": False`` flag to the response information?